### DISCUSSION TITLE
Output points

### DISCUSSION THREAD

**Discussion by beargui** [NOAA-EMC/WW3#1228] [Q&A] [answered]
Hello everyone,

I would really appreciate it if you help me with the following problem:

I am using WW3 v5.16. I have designed a spherical grid covering longitudes both west of the Greenwich meridian and east of it. I would like to obtain wave data information at some locations within my grid (output points). Some of those points are located at the west of the Greenwich meridian and others, at the east. I am wondering which format should have those coordinates to be read by the model in the ww3_shel.inp file.

I have tried the following formats  (_west of Greenwich_, **east of Greenwich**):

Format 1:
    _346.0 53.0 'P001'
    346.0 52.8 'P002'
    346.0 53.6 'P003'_
    **360.0 63.0 'P256'
    360.2 63.0 'P257'
    360.4 63.0 'P258'
    370.0 38.6 'P324'
    370.0 38.4 'P325'
    370.0 38.2 'P326'
    370.0 38.0 'P327'**
     0.0   0.0  'STOPSTRING'

Format 2:
    _346.0 53.0 'P001'
    346.0 52.8 'P002'
    346.0 53.6 'P003'_
    **0.0 63.0 'P256'
    0.2 63.0 'P257'
    0.4 63.0 'P258'
    10.0 38.6 'P324'
    10.0 38.4 'P325'
    10.0 38.2 'P326'
    10.0 38.0 'P327'**
     0.0   0.0  'STOPSTRING'

Format 3:
    _-14.0 53.0 'P001'
    -14.0 52.8 'P002'
    -14.0 53.6 'P003'_
    **0.0 63.0 'P256'
    0.2 63.0 'P257'
    0.4 63.0 'P258'
    10.0 38.6 'P324'
    10.0 38.4 'P325'
    10.0 38.2 'P326'
    10.0 38.0 'P327'**
     0.0   0.0  'STOPSTRING'

Format 4:
    _3.46000000000000000E+02 5.30000000000000000E+01 'P001'
    3.46000000000000000E+02 5.27999999999999972E+01 'P002'
    3.46000000000000000E+02 5.36000000000000014E+01 'P003'_
    **3.60000000000000000E+02 6.30000000000000000E+01 'P256'
    3.60199999999999989E+02 6.30000000000000000E+01 'P257'
    3.60399999999999977E+02 6.30000000000000000E+01 'P258'
    3.70000000000000000E+02 3.85999999999999943E+01 'P324'
    3.70000000000000000E+02 3.83999999999999986E+01 'P325'
    3.70000000000000000E+02 3.82000000000000028E+01 'P326'
    3.70000000000000000E+02 3.80000000000000000E+01 'P327'**
     0.0   0.0  'STOPSTRING'

Format 5:
    _3.46000000000000000E+02 5.30000000000000000E+01 'P001'
    3.46000000000000000E+02 5.27999999999999972E+01 'P002'
    3.46000000000000000E+02 5.36000000000000014E+01 'P003'_
    **0.00000000000000000E+02 6.30000000000000000E+01 'P256'
    0.00199999999999989E+02 6.30000000000000000E+01 'P257'
    0.00399999999999977E+02 6.30000000000000000E+01 'P258'
    0.10000000000000000E+02 3.85999999999999943E+01 'P324'
    0.10000000000000000E+02 3.83999999999999986E+01 'P325'
    0.10000000000000000E+02 3.82000000000000028E+01 'P326'
    0.10000000000000000E+02 3.80000000000000000E+01 'P327'**
     0.0   0.0  'STOPSTRING'

In all of those formats, the model is able to generate spectral information at the points west of the Greenwich meridian. However, I obtain the same error for points from 'P256' to 'P327' (**all the east points**).

 *** WAVEWATCH-III WARNING :
     OUTPUT POINT OUT OF GRID :      0.000    63.000  P256      
     POINT SKIPPPED
 
or

 *** WAVEWATCH-III WARNING :
     OUTPUT POINT OUT OF GRID :    360.000    63.000  P256      
     POINT SKIPPPED

 *** WAVEWATCH-III WARNING :
     OUTPUT POINT OUT OF GRID :    370.000    38.000  P327      
     POINT SKIPPPED

or 

   *** WAVEWATCH-III WARNING :
     OUTPUT POINT OUT OF GRID :     10.000    38.000  P327      
     POINT SKIPPPED

I am sure that those points are covered by my grid.

What I am doing wrong? Why does the model skip only the points east of the Greenwich meridian?

Thank you very much

Bea

**Comment by ukmo-ccbunney** [comment]
@beargui 
That should work ok.
Can you share how you have defined you grid in ww3_grid.inp (or nml)?

**Reply by beargui** [reply]
Of course, here you have

$ -------------------------------------------------------------------- $
$ WAVEWATCH III Grid preprocessor input file                           $
$ -------------------------------------------------------------------- $
  'North Atlantic         '
$
   1.1 0.041772482 25 24 0.
$   1.07 0.035 50 36 0.
   F T T T T T
$
$   900. 300. 450. 10.
$   1800. 900. 1350. 15.
   3600. 900. 3600. 300.
$    3600. 1800. 360. 10.
$
  &MISC FLAGTR = 2 /
$  &SIN4 BETAMAX = 1.33 /
$ &SBT1 GAMMA = -0.019 /
END OF NAMELISTS
$
   'RECT' T 'NONE'
   150  117 
   30.00  30.00  60.00 
   300.0000    7.5000   1.00
$ Bottom Bathymetry 
   -0.10   2.50  40  0.001000  1  1 '(....)'  NAME  'natl.depth' 
$ Sub-grid information 
   50  0.010000  1  1  '(....)'  NAME  'natl.obstr'  
$ Mask Information 
   60  1  1  '(....)'  NAME  'natl.mask'  
$
     0.    0.    0.    0.       0
$ -------------------------------------------------------------------- $
$ End of input file                                                    $
$ -------------------------------------------------------------------- $

**Reply by ukmo-ccbunney** [reply, marked as answer]
@beargui 
I think you are missing the setting that specifies the grid is periodic in longitude space. I.e. this line:
```
<snip>
END OF NAMELISTS
$
'RECT' T 'NONE'    <-------
150 117
</snip>
```

Needs to be this:
```
'RECT' T 'SMPL'
```

`SMPL` means "simple grid closure" and will correctly wrap your longitude values along the prime meridian.

Try regenerating your `mod_def.ww3` file with that new setting in `ww3_grid.inp` and see if that fixes your problem.

**Comment by beargui** [comment]
It seems to be solved now! By writing 'SMPL' as you said. Thank you so much!

**Reply by ukmo-ccbunney** [reply]
Great! I am glad you have got it working.

**Comment by beargui** [comment]
Hello again. 

I thought the problem was then solved by writing 'SMPL' instead of 'NONE', but it wasn't. Some coordinates were misread that way.

Eventually, I was able to find the problem. As initial longitude I should have written -60 instead of 300 in the ww3_grid.inp file. I think the problem is solved now if in the ww3_grid.inp file I write

'RECT' T 'NONE'
150 117
30.00 30.00 60.00
-60.0 7.5000 1.00

and output points in ww3_shel.inp with the following format:

    -14.0 53.0 'P001'
    0.6 63.0 'P259'

Consequently, the present message is just to let us know, just in case we have the same problem in the future. 

Thank you all
